oxide-auth/src/primitives/registrar.rs

```diff
@@ -291,107 +286,69 @@ pub trait PasswordPolicy: Send + Sync {
     fn check(&self, client_id: &str, passphrase: &[u8], stored: &[u8]) -> Result<(), RegistrarError>;
 }
 
-/// Store passwords using `Pbkdf2` to derive the stored value.
-///
-/// Each instantiation generates a 16 byte random salt and prepends this additionally with the
-/// username. This combined string is then used as the salt using the passphrase as the secret to
-/// derive the output. The iteration count defaults to `65536` but can be customized.
-pub struct Pbkdf2 {
-    /// A prebuilt random, or constructing one as needed.
-    random: Option<SystemRandom>,
-    iterations: NonZeroU32,
-}
-
-impl Default for Pbkdf2 {
-    fn default() -> Self {
-        Pbkdf2 {
-            random: Some(SystemRandom::new()),
-            .. *PBKDF2_DEFAULTS
-        }
-    }
+/// Store passwords using `Argon2` to derive the stored value.
+#[derive(Debug)]
+pub struct Argon2<'a> {
+    secret_key: SecretKey<'a>
 }
 
-impl Clone for Pbkdf2 {
-    fn clone(&self) -> Self {
-        Pbkdf2 {
-            random: Some(SystemRandom::new()),
-            .. *self
+impl<'a> Default for Argon2<'a> {
+    fn default() -> Self {   
+        Argon2 {
+            secret_key: "Please provide a SecretKey! This is for your own protection.".into()
```

Comment:
This doesn't really add any protection. This pepper, in terms of password storage, is useful to have but it should either

* Cause an actual error if it is unset.
* Allow being used without a secret key. This is _basically_ what the current function is doing so it won't provide any worse security than at the moment. Would an ephemeral implementation be useful, even though credentials should be long lived?.

Reply:
It is opt-out and would fail if no secret is provided. 
So users should be **required** to provide one?

```
        .opt_out_of_secret_key(true); // Default is `false`
        // 👆 As an extra security measure, if you want to hash without a secret key, which
        // is not recommended, you must explicitly declare that this is your intention
        // by calling this method and setting the `opt_out_of_secret_key` configuration to
        // `true` (by default, it is set to `false`); otherwise hashing will return an error
        // when you fail to provide a secret key
```

Reply:
The crate also documents that the underlying C implementation does not use a secret key for the purpose of password hashing, if I read it correctly. I'd take that as meaning it is fine to opt-out of it.

Reply:
Would Option<SecretKey<'static>> sound good to you?
That way we can trap Some(key) and use it and otherwise opt_out_of_secret_key.

Reply:
That sounds good.

Reply:
Fixed